cmd/gendoc/main.go

```diff
@@ -12,20 +12,26 @@ import (
 	"os/exec"
 	"path/filepath"
 	"regexp"
+	"slices"
 	"sort"
 	"strings"
 	"time"
 )
 
+const maxGathererNameLength = 256
+
 var (
-	inPath          string
-	outPath         string
-	mdf             *os.File
-	randSource      = rand.NewSource(time.Now().UnixNano())
-	reGather        = regexp.MustCompile(`^((Build)?Gather)(.*)`)
-	reExample       = regexp.MustCompile(`^(Example)(.*)`)
-	reSampleArchive = regexp.MustCompile(`docs/(insights-archive-sample/.*)`)
-	cleanRoot       = "./"
+	inPath                   string
+	outPath                  string
+	mdf                      *os.File
+	randSource               = rand.NewSource(time.Now().UnixNano())
+	reGather                 = regexp.MustCompile(`^((Build)?Gather)(.*)`)
+	reExample                = regexp.MustCompile(`^(Example)(.*)`)
+	reSampleArchive          = regexp.MustCompile(`docs/(insights-archive-sample/.*)`)
+	reGathererName           = regexp.MustCompile("(?m)^###\\s*Config ID\\s*\\n`([^`]*)`")
+	reGathererNameValidation = regexp.MustCompile("^[a-z]+[_a-z]*[a-z]([/a-z][_a-z]*)?[a-z]$")
+	gathererNameExceptions   = []string{"ContainersLogs"}
```

Comment:
Is this exception here because of https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/conditional/filter_and_gather_container_logs.go#L51 ?

Reply:
Yes, it has different structure and the regex does not match the gatherer name, because there is none.

Reply:
If it possible to add the name then let's do that